app/controllers/application_controller.rb

```diff
@@ -23,12 +28,23 @@ class ApplicationController < ActionController::Base
     end
   end
 
-  rescue_from EditionAssertions::AccessError do |e|
-    Rails.logger.warn(e.message)
-    render "documents/forbidden", status: :forbidden, assigns: { edition: e.edition }
-  end
-
   def rendering_context
     request.headers["Content-Publisher-Rendering-Context"] || "application"
   end
+
+  def check_user_access
+    document_id = request.path_parameters[:document]
+    return if document_id.blank?
+
+    access_limit = Edition.includes(:access_limit)
```

Comment:
It'd be good to have this join also include `revision: :tags_revision` to avoid the extra query for that.